edx_management_commands/management_commands/management/commands/put_group.py

```diff
@@ -0,0 +1,118 @@
+
+    @transaction.atomic
+    def handle(self, group_name, permissions=None, *args, **options):
+
+        old_permissions, new_permissions = [], []
+        group, created = Group.objects.get_or_create(name=group_name)  # pylint: disable=no-member
+
+        if created:
+            try:
+                # Needed for sqlite backend (i.e. in tests) because
+                # name.max_length won't be enforced by the db.
+                # See also http://www.sqlite.org/faq.html#q9
+                group.full_clean()
+            except ValidationError, exc:
+                # give a more helpful error
+                raise CommandError(
+                    _(
+                        'Invalid group name: "{group_name}". {messages}'
+                    ).format(
+                        group_name=group_name,
+                        messages=exc.messages[0]
+                    )
+                )
+            self.stderr.write(_('Created new group: "{}"').format(group_name))
+        else:
+            self.stderr.write(_('Found existing group: "{}"').format(group_name))
+            old_permissions = list(group.permissions.all())
+
+        # resolve the specified permissions
+        for permission in permissions or []:
+
+            try:
+                app_label, model_name, codename = permission.split(':')
+            except ValueError:
+                # give a more helpful error
+                raise CommandError(_(
+                    'Invalid permission option: "{}". Please specify permissions '
+                    'using the format: app_label:model_name:permission_codename.'
+                ).format(permission))
+            # this will raise a LookupError if it fails.
+            try:
+                model_class = apps.get_model(app_label, model_name)
+            except LookupError, exc:
+                raise CommandError(exc.message)
+
+            content_type = ContentType.objects.get_for_model(model_class)
+            try:
+                new_permission = Permission.objects.get(  # pylint: disable=no-member
+                    content_type=content_type,
+                    codename=codename,
+                )
+            except Permission.DoesNotExist:
+                # give a more helpful error
+                raise CommandError(
+                    _(
+                        'Invalid permission codename: "{codename}".  No such permission exists '
+                        'for the model {module}.{model_name}.'
+                    ).format(
+                        codename=codename,
+                        module=model_class.__module__,
+                        model_name=model_class.__name__,
+                    )
+                )
+            new_permissions.append(new_permission)
+
+        add_permissions = [p for p in new_permissions if p not in old_permissions]
+        remove_permissions = [p for p in old_permissions if p not in new_permissions]
```

Comment:
Can we add help text to the web ui that discusses the dangers of manual changes?  Other choice might be to fail if len(remove_permissions) > 0 and have a --force-removal option.